packages/components/src/Form/Fields/FieldInline.tsx

```diff
@@ -54,20 +54,22 @@ const FieldInlineLayout: FC<FieldInlinePropsInternal> = ({
   validationMessage,
 }) => {
   return (
-    <label className={className} htmlFor={id}>
-      <InputArea>{children}</InputArea>
-      <Label as="span">
+    <div className={className}>
+      <InputArea aria-describedby={id}>{children}</InputArea>
+      <Label id={id} as="span">
         <Truncate>{label}</Truncate>
         {required && <RequiredStar />}
       </Label>
-      {detail && <FieldDetail>{detail}</FieldDetail>}
-      <MessageArea id={`${id}-describedby`}>
-        {description && <FieldDescription>{description}</FieldDescription>}
+      {detail && <FieldDetail id={id}>{detail}</FieldDetail>}
```

Comment:
Not sure how to attach the detail to the input but `id={id}` will definitely bust things.

```suggestion
      {detail && <FieldDetail id={`${id}-detail`}>{detail}</FieldDetail>}
```